packages/core/src/oidc/init.ts

```diff
@@ -117,8 +121,14 @@ export default async function initOidc(app: Koa): Promise<Provider> {
     },
   });
 
+  /**
+   * OIDC provider listeners and events
+   * https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#im-getting-a-client-authentication-failed-error-with-no-details
+   * https://github.com/panva/node-oidc-provider/blob/v7.x/docs/events.md
+   */
   oidc.on('grant.success', grantSuccessListener);
   oidc.on('grant.error', grantErrorListener);
+  oidc.on('grant.revoked', grantRevokedListener);
```

Comment:
nit: could extract and wrapped up these into a `registerOIDCEventListener(oidc)` in your utils file.

Reply:
Refactored. PTAL

```ts
export const addOidcEventListeners = (provider: Provider) => {
  /**
   * OIDC provider listeners and events
   * https://github.com/panva/node-oidc-provider/blob/main/docs/README.md#im-getting-a-client-authentication-failed-error-with-no-details
   * https://github.com/panva/node-oidc-provider/blob/v7.x/docs/events.md
   */
  provider.addListener('grant.success', grantSuccessListener);
  provider.addListener('grant.error', grantErrorListener);
  provider.addListener('grant.revoked', grantRevokedListener);
};
```